capsules/src/nonvolatile_storage_driver.rs

```diff
@@ -526,46 +586,54 @@ impl LegacyDriver for NonvolatileStorage<'_> {
     /// - `1`: Return the number of bytes available to userspace.
     /// - `2`: Start a read from the nonvolatile storage.
     /// - `3`: Start a write to the nonvolatile_storage.
-    fn command(&self, arg0: usize, arg1: usize, _: usize, appid: AppId) -> ReturnCode {
-        let command_num = arg0 & 0xFF;
-
+    fn command(
+        &self,
+        command_num: usize,
+        offset: usize,
+        length: usize,
+        appid: AppId,
+    ) -> CommandResult {
         match command_num {
-            0 =>
-            /* This driver exists. */
-            {
-                ReturnCode::SUCCESS
+            0 /* This driver exists. */ => {
+                CommandResult::success()
             }
 
-            // How many bytes are accessible from userspace.
-            1 => ReturnCode::SuccessWithValue {
-                value: self.userspace_length,
+            1 /* How many bytes are accessible from userspace */ => {
+		// TODO: Would break on 64-bit platforms
```

Comment:
well, only if more than 4 GB are reserved as accessible from userspace, which seems like a platform pretty far outside Tock's wheelhouse for now